File: js-test/servers/common/core-plugins/tests/1/as/webapps/openbis-v3-api-test/html/models/dto/search/AbstractEntitySearchCriterion.js
/**
 * @author pkupczyk
 */
define([ "support/stjs", "dto/search/AbstractObjectSearchCriterion", "dto/search/SearchOperator", "dto/search/CodeSearchCriterion", "dto/search/EntityTypeSearchCriterion",
		"dto/search/PermIdSearchCriterion", "dto/search/RegistrationDateSearchCriterion", "dto/search/ModificationDateSearchCriterion", "dto/search/TagSearchCriterion",
		"dto/search/StringPropertySearchCriterion", "dto/search/DatePropertySearchCriterion", "dto/search/AnyPropertySearchCriterion", "dto/search/AnyFieldSearchCriterion",
		"dto/search/AbstractCompositeSearchCriterion" ], function(stjs, AbstractObjectSearchCriterion, SearchOperator, CodeSearchCriterion, EntityTypeSearchCriterion, PermIdSearchCriterion,
		RegistrationDateSearchCriterion, ModificationDateSearchCriterion, TagSearchCriterion, StringPropertySearchCriterion, DatePropertySearchCriterion, AnyPropertySearchCriterion,
		AnyFieldSearchCriterion, AbstractCompositeSearchCriterion) {
	var AbstractEntitySearchCriterion = function() {
		AbstractObjectSearchCriterion.call(this);
	};
	stjs.extend(AbstractEntitySearchCriterion, AbstractObjectSearchCriterion, [ AbstractObjectSearchCriterion ], function(constructor, prototype) {
		prototype['@type'] = 'dto.search.AbstractEntitySearchCriterion';
		constructor.serialVersionUID = 1;
		prototype.operator = SearchOperator.AND;
		prototype.withOperator = function(anOperator) {
			this.operator = anOperator;
			return this;
		};
		prototype.getOperator = function() {
			return this.operator;
		};
		prototype.withCode = function() {
			return this.addCriterion(new CodeSearchCriterion());
		};
		prototype.withType = function() {
			return this.addCriterion(new EntityTypeSearchCriterion());
		};
		prototype.withPermId = function() {
			return this.addCriterion(new PermIdSearchCriterion());
		};
		prototype.withRegistrationDate = function() {
			return this.addCriterion(new RegistrationDateSearchCriterion());
		};
		prototype.withModificationDate = function() {
			return this.addCriterion(new ModificationDateSearchCriterion());
		};
		prototype.withTag = function() {
			return this.addCriterion(new TagSearchCriterion());
		};
		prototype.withProperty = function(propertyName) {
			return this.addCriterion(new StringPropertySearchCriterion(propertyName));
		};
		prototype.withDateProperty = function(propertyName) {
			return this.addCriterion(new DatePropertySearchCriterion(propertyName));
		};
		prototype.withAnyProperty = function() {
			return this.addCriterion(new AnyPropertySearchCriterion());
		};
		prototype.withAnyField = function() {
			return this.addCriterion(new AnyFieldSearchCriterion());
		};
		prototype.createBuilder = function() {
			var builder = AbstractCompositeSearchCriterion.prototype.createBuilder.call(this);
			builder.setOperator(this.operator);
			return builder;
		};
	}, {
		operator : {
			name : "Enum",
			arguments : [ "SearchOperator" ]
		},
		criteria : {
			name : "Collection",
			arguments : [ "ISearchCriterion" ]
		}
	});
	return AbstractEntitySearchCriterion;
})

File: openbis_api/source/java/ch/ethz/sis/openbis/generic/shared/api/v3/dto/search/AbstractEntitySearchCriterion.java
/*
 * Copyright 2014 ETH Zuerich, Scientific IT Services
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.shared.api.v3.dto.search;

import ch.ethz.sis.openbis.generic.shared.api.v3.dto.id.IObjectId;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author pkupczyk
 */
@JsonObject("dto.search.AbstractEntitySearchCriterion")
public class AbstractEntitySearchCriterion<ID extends IObjectId> extends AbstractObjectSearchCriterion<ID>
{

    private static final long serialVersionUID = 1L;

    public CodeSearchCriterion withCode()
    {
        return with(new CodeSearchCriterion());
    }

    public EntityTypeSearchCriterion withType()
    {
        return with(new EntityTypeSearchCriterion());
    }

    public PermIdSearchCriterion withPermId()
    {
        return with(new PermIdSearchCriterion());
    }

    public RegistrationDateSearchCriterion withRegistrationDate()
    {
        return with(new RegistrationDateSearchCriterion());
    }

    public ModificationDateSearchCriterion withModificationDate()
    {
        return with(new ModificationDateSearchCriterion());
    }

    public TagSearchCriterion withTag()
    {
        return with(new TagSearchCriterion());
    }

    public StringPropertySearchCriterion withProperty(String propertyName)
    {
        return with(new StringPropertySearchCriterion(propertyName));
    }

    public DatePropertySearchCriterion withDateProperty(String propertyName)
    {
        return with(new DatePropertySearchCriterion(propertyName));
    }

    public AnyPropertySearchCriterion withAnyProperty()
    {
        return with(new AnyPropertySearchCriterion());
    }

    public AnyFieldSearchCriterion withAnyField()
    {
        return with(new AnyFieldSearchCriterion());
    }

    @Override
    protected SearchCriterionToStringBuilder createBuilder()
    {
        SearchCriterionToStringBuilder builder = super.createBuilder();
        builder.setOperator(operator);
        return builder;
    }

}
